with-better-auth/app/(auth)/sign-up.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState } from "react";
+import { TextInput, Button, ScrollView, Text, StyleSheet } from "react-native";
+import { authClient } from "@/lib/auth-client";
+
+export default function SignUp() {
+  const [email, setEmail] = useState("");
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+
+  const handleSignUp = async () => {
+    await authClient.signUp.email({
+      email,
+      password,
+      name,
+    });
+  };
+
+  return (
+    <ScrollView contentInsetAdjustmentBehavior="automatic">
+      <Text style={{ fontSize: 32, fontWeight: "bold", margin: 10 }}>
+        Sign Up
+      </Text>
+      <TextInput
+        placeholder="Name"
+        value={name}
+        onChangeText={setName}
+        style={styles.input}
+      />
+      <TextInput
+        placeholder="Email"
+        value={email}
+        onChangeText={setEmail}
+        style={styles.input}
+      />
```

Comment:
Nit - maybe add following props; 
- inputMode
- autoCapitalize

```suggestion
<TextInput
  placeholder="Email"
  value={email}
  inputMode="email"
  autoCapitalize="none"
  onChangeText={setEmail}
  style={styles.input}
/>
```